Source/WarpX.cpp

```diff
@@ -1176,52 +1178,57 @@ WarpX::ReadParameters ()
         // Check whether the default comoving velocity should be used
         bool use_default_v_comoving = false;
         pp_psatd.query("use_default_v_comoving", use_default_v_comoving);
+
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            !use_default_v_comoving || boosted_frame,
+            "psatd.use_default_v_comoving = 1 can be used only if warpx.gamma_boost is also set"
+        );
+
         if (use_default_v_comoving == true && boosted_frame == true)
         {
             m_v_comoving[2] = -std::sqrt(1._rt - 1._rt / (gamma_boost * gamma_boost));
         }
-        else if (use_default_v_comoving == true && boosted_frame == false)
-        {
-            amrex::Abort("psatd.use_default_v_comoving = 1 can be used only if warpx.gamma_boost is also set");
-        }
         else
         {
             queryArrWithParser(pp_psatd, "v_comoving", m_v_comoving, 0, 3);
         }
 
-        // Galilean and comoving algorithms should not be used together
-        if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.)
-        {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.)
-            {
-                amrex::Abort("Galilean and comoving algorithms should not be used together");
-            }
-        }
-
         // Scale the Galilean/comoving velocity by the speed of light
-        for (int i=0; i<3; i++) m_v_galilean[i] *= PhysConst::c;
-        for (int i=0; i<3; i++) m_v_comoving[i] *= PhysConst::c;
+        for (auto& vv : m_v_galilean) vv*= PhysConst::c;
+        for (auto& vv : m_v_comoving) vv*= PhysConst::c;
+
+        const auto v_galilean_is_zero =
+            std::all_of(m_v_galilean.begin(), m_v_galilean.end(),
+            [](const auto& val){return val == 0.;});
+
+        const auto v_comoving_is_zero =
+            std::all_of(m_v_comoving.begin(), m_v_comoving.end(),
+            [](const auto& val){return val == 0.;});
+
+
+        // Galilean and comoving algorithms should not be used together
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            v_galilean_is_zero || v_comoving_is_zero,
+            "Galilean and comoving algorithms should not be used together"
+        );
 
         // The comoving PSATD algorithm is not implemented nor tested with Esirkepov current deposition
-        if (current_deposition_algo == CurrentDepositionAlgo::Esirkepov) {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.) {
-                amrex::Abort("Esirkepov current deposition cannot be used with the comoving PSATD algorithm");
-            }
-        }
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            (current_deposition_algo != CurrentDepositionAlgo::Esirkepov) ||
```

Comment:
I think this also needs to be an `xor`
```suggestion
            (current_deposition_algo == CurrentDepositionAlgo::Esirkepov) ^
```

Reply:
Uhm, I am not sure.. not (A and B) shouldn't be (not A) or (not B) ? 

Here, we should abort only when Esirkepov is used together with the co-moving algorithm.

Using `assert ( (dep != Esirkepov) || (v_comoving==0) )` the truth  table is:

| deposition   | v_comoving | result |
|--------------|------------|--------|
| Esirkepov    | 0          | OK     |
| Esirkepov    | != 0       | ABORT  |
| != Esirkepov | 0          | OK     |
| != Esirkepov | != 0       | OK     |

while using `assert ( (dep == Esirkepov) XOR (v_comoving==0) )` the truth  table is:

| deposition   | v_comoving | result |
|--------------|------------|--------|
| Esirkepov    | 0          | ABORT |
| Esirkepov    | != 0       | OK  |
| != Esirkepov | 0          | OK     |
| != Esirkepov | != 0       | ABORT     |

which is not what we want...